Source/WebCore/Modules/WebGPU/GPUBindGroupDescriptor.h

```diff
@@ -46,6 +46,32 @@ struct GPUBindGroupDescriptor : public GPUObjectDescriptorBase {
         };
     }
 
+    const RefPtr<GPUExternalTexture>* externalTextureMatches(Vector<GPUBindGroupEntry>& comparisonEntries, bool& hasExternalTexture) const
+    {
+        bool matched = true;
+        hasExternalTexture = false;
+        auto entriesSize = entries.size();
+        if (entriesSize != comparisonEntries.size())
+            matched = false;
+
+        const RefPtr<GPUExternalTexture>* result = nullptr;
+        for (size_t i = 0; i < entriesSize; ++i) {
+            auto& entry = entries[i];
+            if (matched && !GPUBindGroupEntry::equal(entry, comparisonEntries[i]))
+                matched = false;
+
+            auto externalTexture = entry.externalTexture();
+            if (!result)
+                result = externalTexture;
+            else if (externalTexture)
+                return nullptr;
+            if (result)
+                hasExternalTexture = true;
```

Comment:
We can't early return because we want this variable to be set even when the match result is false @tadeuzagallo 

But maybe there is a better way to write the code since I agree it is somewhat confusing